### DISCUSSION TITLE
Where is experiment-generated audio stored?

### DISCUSSION THREAD

**Discussion by kjkendro** [jspsych/jsPsych#1202] [Q&A] [answered]
I'm very new to coding, so apologies if this has a very obvious answer that I'm not seeing!

I'm using the jspsych-html-audio-response plugin, and I'm looking to run an MTurk study to collect audio data from participants during a picture naming task. Is there a default place where the audio files are stored after they are generated by jsPsych using this plugin? Is there a way to send them somewhere specific (such as DropBox)? 

Thank you for any guidance you can provide!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @kjkendro, great question, and sorry for the lack of documentation on this. The audio-response plugins have not been officially added to the jsPsych library yet, but when they're added we'll include information about this in the docs!

By default, the audio response data is stored as a base 64 string in the jsPsych data. This allows the audio data to be stored as a string, along with the rest of the jsPsych data, so that you don't need to do anything 'extra' to save the responses. Once you have the audio data as a base 64 string, you can convert to an audio file using R ([example code here](https://github.com/jspsych/jsPsych/discussions/796#discussioncomment-33705)) or Python ([example code here](https://github.com/jspsych/jsPsych/issues/494#issuecomment-705674385)), or using an [online base64-to-audio converter](https://base64.guru/converter/decode/audio) (this is useful for testing, but I don't recommend using this with data from real participants for data privacy reasons). 

However, even short audio recordings will produce very very long base 64 strings , i.e. thousands of characters, which could cause data transfer/storage problems if you have lots of and/or very long recordings during your task. So if possible, I suggest saving the audio data as a file on your server using the plugin's `postprocessing` parameter. See [here](https://github.com/jspsych/jsPsych/issues/494#issuecomment-725052229) for an example of how I've used the `postprocessing` function to send the audio files back to a JATOS server. The plugin files also contain a [description of the postprocessing parameter](https://github.com/becky-gilbert/jsPsych/blob/fbc51f44824023b488ad88d34c0febec2625f8fa/plugins/jspsych-image-audio-response.js#L63) and the [default function](https://github.com/becky-gilbert/jsPsych/blob/fbc51f44824023b488ad88d34c0febec2625f8fa/plugins/jspsych-image-audio-response.js#L37), which might be helpful for figuring out how to use it.
